### PR TITLE
Revert Refs back to Array for TOL in arpack.jl

### PR DESCRIPTION
**Pull request by musm**
#TODO change back after 0.6 is branched and update wrapper

### CODE REVIEW

File: base/linalg/arpack.jl [hunk 1]
```diff
@@ -257,7 +257,7 @@ for (T, TR, naupd_name, neupd_name) in
     ((:Complex128, :Float64, :znaupd_, :zneupd_),
      (:Complex64,  :Float32, :cnaupd_, :cneupd_))
     @eval begin
-        function naupd(ido, bmat, n, evtype, nev, TOL::Ref{$TR}, resid::Array{$T}, ncv, v::Array{$T}, ldv,
+        function naupd(ido, bmat, n, evtype, nev, TOL::Union{Array{$T},Ref{$T}}, resid::Array{$T}, ncv, v::Array{$T}, ldv,
```
Comment (tkelman) [inline base/linalg/arpack.jl]: TR ?

Comment (musm) [inline base/linalg/arpack.jl]: what's TR?

**Comment by tkelman** [comment]
also eupd_wrapper

File: base/linalg/arpack.jl (hunk 1 above)
Comment (tkelman) [inline base/linalg/arpack.jl]: the type parameter

Comment (musm) [inline base/linalg/arpack.jl]: I'm still confused here, care to clarify?

**Comment by musm** [comment]
fixed

File: base/linalg/arpack.jl (hunk 1 above)
Comment (pabloferz) [inline base/linalg/arpack.jl]: I belive Tony means that the type parameter here should be `TR` and not just `T` (you can see `TR` is being used in the `ccall` below.

**Comment by andreasnoack** [comment]
See https://github.com/JuliaLang/julia/pull/21420#discussion_r112807532

**Comment by musm** [comment]
reverted

**Pull request merged**